### DISCUSSION TITLE
Hibernate L2C + Jcache

### DISCUSSION THREAD

**Discussion by UbiquitousBear** [infinispan/infinispan#11631] [Q&A]
I'm using Infinispan via Jcache to provide L2C support to Hibernate. I've enabled query cache. 

When querying, I'm receiving a marshal exception:  `No marshaller registered for object of Java type org.hibernate.cache.spi.QueryKey : org.hibernate.cache.spi.QueryKey@539e5071. `

How do I configure the serialization/marshaller for QueryKey?

**Comment by tristantarrant** [comment]
Why are you not using our native Hibernate 2LC provider instead ? https://infinispan.org/docs/stable/titles/hibernate/hibernate.html

**Reply by UbiquitousBear** [reply]
I believe that uses the embedded cache and not the remote cache?
